TemplateOperations/Difference/TemplateDifference.wl

```diff
@@ -0,0 +1,79 @@
+(* ::Package:: *)
+
+BeginPackage[
+  "CATemplates`TemplateOperations`Difference`TemplateDifference`",
+  {
+    "CATemplates`Basic`",
+    "CATemplates`CATemplate`",
+    "CATemplates`TemplateOperations`Expansion`PostExpansionFn`IdentityFn`",
+    "CATemplates`TemplateOperations`Expansion`PostExpansionFn`FilterNotAllowed`",
+    "CATemplates`TemplateOperations`Expansion`PostExpansionFn`FilterOutOfRange`",
+    "CATemplates`TemplateOperations`Difference`Common`",
+    "CATemplates`TemplateOperations`Difference`RawDifference`",
+    "CATemplates`TemplateOperations`Difference`ExceptionTemplates`"}
+];
+
+
+TemplateDifference::usage = "bl1";
+TemplateDifferenceDirect::usage = "bl1";
+ExpansionFunctionSelected::usage = "bl1";
+
+Begin["`Private`"];
+
+ExpansionFunctionSelected[x_, y_] := Switch[{x, y},
+  (*{IdentityFn, IdentityFn}, FilterOutOfRange,*)
+  {IdentityFn, FilterOutOfRange}, FilterOutOfRange,
+  _, x
+];
+
+ExpansionFunctionSelected[expansion_] := With[{result = expansion},
+  If[result === RawExpansion,
+    result = FilteredExpansion;
+  ];
+  result
+];
+
+TemplateDifference[template1_Association, template2_Association] :=
+    With[{
+      k = template1[["k"]],
+      r = template1[["r"]],
+      n = template1[["N"]],
```

Comment:
Use accessor functions defined in `CATemplate` instead of accessing template's fields directly.